hbase-client/src/main/java/org/apache/hadoop/hbase/MetaRegionsNotAvailableException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase;
+
+import org.apache.yetus.audience.InterfaceAudience;
+
+import java.io.IOException;
+
+/**
+ * Thrown by master when meta region locations are not cached whatever reason.
+ * Client is expected to retry when running into this.
+ */
+@InterfaceAudience.Private
+public class MetaRegionsNotAvailableException extends IOException  {
```

Comment:
Is there going to be other information served by this registry beside meta regions? Maybe not right now but do you think this mechanism may be overloaded in the future? It seems generally useful... masters as intermediary between clients (or servers) and what is direct access to zookeeper today. It would be better to not have a checked exception for each bit of information served via this means. Would IOException be sufficient?